tests/csapi/txnid_test.go

```diff
@@ -22,20 +25,192 @@ func TestTxnInEvent(t *testing.T) {
 	// Create a room where we can send events.
 	roomID := c.CreateRoom(t, map[string]interface{}{})
 
+	txnId := "abcdefg"
 	// Let's send an event, and wait for it to appear in the timeline.
-	eventID := c.SendEventSynced(t, roomID, b.Event{
+	eventID := c.SendEventUnsyncedWithTxnID(t, roomID, b.Event{
 		Type: "m.room.message",
 		Content: map[string]interface{}{
 			"msgtype": "m.text",
 			"body":    "first",
 		},
-	})
+	}, txnId)
 
 	// The transaction ID should be present on the GET /rooms/{roomID}/event/{eventID} response.
 	res := c.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "rooms", roomID, "event", eventID})
 	body := client.ParseJSON(t, res)
 	result := gjson.ParseBytes(body)
-	if !result.Get("unsigned.transaction_id").Exists() {
-		t.Fatalf("Event did not have a 'transaction_id' on the GET /rooms/%s/event/%s response", roomID, eventID)
+	unsignedTxnId := result.Get("unsigned.transaction_id")
+	if !unsignedTxnId.Exists() {
+		t.Fatalf("Event did not have a 'unsigned.transaction_id' on the GET /rooms/%s/event/%s response", roomID, eventID)
+	}
+
+	must.EqualStr(t, unsignedTxnId.Str, txnId, fmt.Sprintf("Event did not have a 'unsigned.transaction_id' on GET /rooms/%s/event/%s response", eventID, roomID))
```

Comment:
I think this error message should say "had an incorrect transaction id" rather than "did not have a  transaction id"? Similarly for line 59 below